File: src/main/java/engineer/thomas_werner/euler/Problem6.java
package engineer.thomas_werner.euler;

public class Problem6 {

    public static void main(String[] args) {
        final Problem6 problem6 = new Problem6();
        final long result = Math.abs(problem6.getSumOfSquares(100) - problem6.getSquareOfSums(100));
        System.out.println(result);
    }

    public long getSumOfSquares(final int limit) {
        long result = 0;
        for(int i=1; i<= limit; i++)
            result += i*i;
        return result;
    }

    public long getSquareOfSums(final int limit) {
        long sum = 0;
        for(int i=1; i<= limit; i++)
            sum += i;
        return sum * sum;
    }

}

File: src/main/kotlin/de/huddeldaddel/euler/Problem6.kt
package de.huddeldaddel.euler

/**
 * Solution for https://projecteuler.net/problem=6
 */
fun main() {
    val problem6 = Problem6()
    val result = Math.abs(problem6.getSumOfSquares(100) - problem6.getSquareOfSums(100))
    println(result)
}

class Problem6 {

    fun getSumOfSquares(limit: Int): Long {
        var result: Long = 0
        for (i in 1..limit)
            result += (i * i).toLong()
        return result
    }

    fun getSquareOfSums(limit: Int): Long {
        var sum: Long = 0
        for (i in 1..limit)
            sum += i.toLong()
        return sum * sum
    }

}
